docs/research-branch-policy.md

```diff
@@ -0,0 +1,57 @@
+# Research Branch Policy for InstructLab
+
+This document proposes a research branch policy for InstructLab repositories (sdg, eval, training) in the InstructLab organization.
+
+## Background
+
+Research engineers contributing to InstructLab have expressed the need for the ability to make rapid, experimental changes in a space where they can collaborate and move quickly without working through robust, production-level code review. 
+Software engineers contributing to InstructLab have expressed concern about allowing code into the `main` branch that does not meet the production-level quality standards enforced on main. We need to minimize the time from code merged into main to shipping in a release, and automating confidence in that shipment.
+
+One of the more unique considerations InstructLab has as a project in contrast to other open source projects is the tight collaboration between research engineers and software engineers. There are cultural differences and different best practices in each discipline, and in order to enable this partnership we may need to take some new or less standard approaches to open source project management.
+
+### Current temporary solution
+
+A temporary working solution to enable research engineers to make forward progress is creating a `research` branch in a fork of the InstructLab repository under a user account. This enables the researchers to move forward rapidly and experiment in a collaborative manner. This will also necessitate a more involved merge process once the branch is ready to merge into the `main` branch of the `instructlab` repo under the InstructLab GitHub organization. That is a known and accepted tradeoff of not doing continuous production-level code review.
+
+### Proposed long-term solution
+
+Longer-term, however, the researchers have expressed concern that having this `research` repository in a user's fork makes it more difficult for would-be research-oriented contributors to discover. They would like to have a `research` branch (akin to the common practice of having a `devel` branch) on the main `instructlab` repositories such that their work is discoverable in addition to not negatively impacting the corresponding production-quality `main` branch. 
```

Comment:
Could you please share one or more projects that use a `devel` branch besides their `main` branch? Thanks!

Reply:
https://nvie.com/posts/a-successful-git-branching-model/

any project that uses the gitflow branching strategy?

Reply:
Perhaps we need an `instructlab-research` GitHub organization where research forks of the main `instructlab` repos are kept. This would be a central place for people to discover ongoing InstructLab research efforts and using forks there will facilitate making pull requests back to the main repos when ready to drive the research work back into production code. 

Having "research" branches in the main repos does not make them discoverable.

Reply:
@bjhargrave I actually really like this idea! what are the main downsides?

Reply:
The only downside I can think of is having to manage another GH organization. But that is also a feature. We probably need to have different maintainers for production work than for research work. Using branches in a single repo makes things complex (probably requires CODEOWNERS which vary by branch which is a PITA). With a separate GH org, you can manage the teams and their application to research repos in a more straightforward and common manner perhaps with more relaxed rules for who can be a committer in research repos than you would want for production repos.